Source/PacketizedTCP.cpp

```diff
@@ -87,8 +87,8 @@ bool PacketizedTCP::SendList( const char **data, const unsigned int *lengths, co
 #endif
 	
 
-	unsigned int lengthsArray[512];
-	const char *dataArray[512];
+	unsigned int lengthsArray[513];
+	const char *dataArray[513];

```

Comment:
As @kfsone pointed out in his comment, increasing the array size is unnecessary. See the condition in the for-loop in line 94 which explicitly checks for i < 512 (in addition to i < numParameters). Not a common coding style, I'd say (and therefore easily to overlook) but this should prevent any out of bounds access, no?